pkg/engine/internal/planner/physical/planner.go

```diff
@@ -213,6 +213,15 @@ func (p *Planner) processMakeTable(lp *logical.MakeTable, ctx *Context) (Node, e
 	p.plan.graph.Add(scanSet)
 
 	for _, dataObj := range dataObjs {
+		predicatesToPushdown := make([]Expression, 0, len(predicates))
+		if len(dataObj.Labels) > 0 {
+			for _, predicate := range predicates {
+				p, disambiguated := disambiguateExpression(predicate, dataObj.Labels)
```

Comment:
i think the disambiguity has to be at a stream-level, not section-wide.

assume the section has logs from 2 streams:
stream A with labels `foo, bar`
stream B with labels `bar, buzz` 

we now assume these 3 are stream labels for all rows in the section. predicate `buzz=something` which now operates on labels would filter out all rows from stream A even though Stream A logs could contain `buzz` as structured metadata